Content.Server/_Goobstation/Heretic/EntitySystems/HereticSystem.cs

```diff
@@ -147,9 +147,12 @@ private void OnRerollTargets(Entity<HereticComponent> ent, ref EventHereticRerol
         }
 
         // add whatever more until satisfied
-        for (int i = 0; i <= ent.Comp.MaxTargets - pickedTargets.Count; i++)
-            if (eligibleTargets.Count > 0)
-                pickedTargets.Add(_rand.PickAndTake<EntityUid>(eligibleTargets));
+        while (ent.Comp.MaxTargets - pickedTargets.Count > 0)
```

Comment:
would `ent.Comp.MaxTargets > pickedTargets.Count` not be eeever so slightly faster? its pretty inconsequential, but while we're making improvements..

good catch on this, though